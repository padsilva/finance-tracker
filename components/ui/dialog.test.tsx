import { Cross2Icon } from "@radix-ui/react-icons";
import { render } from "@testing-library/react";

import {
  Dialog,
  DialogOverlay,
  DialogTrigger,
  DialogClose,
  DialogContent,
  DialogHeader,
  DialogFooter,
  DialogTitle,
  DialogDescription,
} from "./dialog";

// Mock portal to avoid ReactDOM.createPortal issues in tests
jest.mock("@radix-ui/react-dialog", () => ({
  Root: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="dialog-root">{children}</div>
  ),
  Trigger: ({ children }: { children: React.ReactNode }) => (
    <button data-testid="dialog-trigger">{children}</button>
  ),
  Portal: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="dialog-portal">{children}</div>
  ),
  Overlay: ({
    children,
    className,
  }: {
    children?: React.ReactNode;
    className?: string;
  }) => (
    <div data-testid="dialog-overlay" className={className}>
      {children}
    </div>
  ),
  Content: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) => (
    <div data-testid="dialog-content" className={className}>
      {children}
    </div>
  ),
  Close: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) => (
    <button data-testid="dialog-close" className={className}>
      {children}
    </button>
  ),
  Title: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) => (
    <h2 data-testid="dialog-title" className={className}>
      {children}
    </h2>
  ),
  Description: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) => (
    <p data-testid="dialog-description" className={className}>
      {children}
    </p>
  ),
}));

describe("Dialog Components", () => {
  it("should render basic dialog correctly", () => {
    const { container } = render(
      <Dialog>
        <DialogTrigger>Open Dialog</DialogTrigger>
        <DialogContent>
          <DialogHeader>
            <DialogTitle>Dialog Title</DialogTitle>
            <DialogDescription>Dialog Description</DialogDescription>
          </DialogHeader>
          <div>Dialog Content</div>
          <DialogFooter>
            <DialogClose>Close Dialog</DialogClose>
          </DialogFooter>
        </DialogContent>
      </Dialog>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render dialog with custom styling", () => {
    const { container } = render(
      <Dialog>
        <DialogTrigger className="custom-trigger">Open</DialogTrigger>
        <DialogContent className="custom-content">
          <DialogHeader className="custom-header">
            <DialogTitle className="custom-title">Custom Title</DialogTitle>
            <DialogDescription className="custom-description">
              Custom Description
            </DialogDescription>
          </DialogHeader>
          <DialogFooter className="custom-footer">
            <DialogClose className="custom-close">Close</DialogClose>
          </DialogFooter>
        </DialogContent>
      </Dialog>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render dialog with complex content", () => {
    const { container } = render(
      <Dialog>
        <DialogTrigger>Open Settings</DialogTrigger>
        <DialogContent>
          <DialogHeader>
            <DialogTitle>Settings</DialogTitle>
            <DialogDescription>
              Manage your account settings and preferences.
            </DialogDescription>
          </DialogHeader>
          <div className="grid gap-4 py-4">
            <div className="grid grid-cols-4 items-center gap-4">
              <label htmlFor="name">Name</label>
              <input id="name" className="col-span-3" />
            </div>
            <div className="grid grid-cols-4 items-center gap-4">
              <label htmlFor="username">Username</label>
              <input id="username" className="col-span-3" />
            </div>
          </div>
          <DialogFooter>
            <DialogClose>Cancel</DialogClose>
            <button type="submit">Save changes</button>
          </DialogFooter>
        </DialogContent>
      </Dialog>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render dialog overlay with animations", () => {
    const { container } = render(
      <DialogOverlay className="custom-overlay" data-state="open" />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render dialog with different states", () => {
    const { container } = render(
      <Dialog>
        <DialogContent data-state="open">
          <DialogTitle>Open State</DialogTitle>
        </DialogContent>
        <DialogContent data-state="closed">
          <DialogTitle>Closed State</DialogTitle>
        </DialogContent>
      </Dialog>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render header with different alignments", () => {
    const { container } = render(
      <>
        <DialogHeader>
          <DialogTitle>Default Centered Header</DialogTitle>
        </DialogHeader>
        <DialogHeader className="sm:text-left">
          <DialogTitle>Left Aligned Header</DialogTitle>
        </DialogHeader>
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render footer with different layouts", () => {
    const { container } = render(
      <>
        <DialogFooter>
          <button>Cancel</button>
          <button>Confirm</button>
        </DialogFooter>
        <DialogFooter className="flex-row justify-between">
          <button>Back</button>
          <div className="flex gap-2">
            <button>Cancel</button>
            <button>Next</button>
          </div>
        </DialogFooter>
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with close button and icon", () => {
    const { container } = render(
      <DialogContent>
        <DialogTitle>Dialog with Close</DialogTitle>
        <DialogDescription>Content here</DialogDescription>
        <DialogClose>
          <Cross2Icon className="h-4 w-4" />
          <span className="sr-only">Close</span>
        </DialogClose>
      </DialogContent>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render responsive dialog content", () => {
    const { container } = render(
      <DialogContent className="max-w-[90%] md:max-w-[80%] lg:max-w-[60%]">
        <DialogHeader>
          <DialogTitle>Responsive Dialog</DialogTitle>
        </DialogHeader>
        <div className="grid grid-cols-1 gap-4 md:grid-cols-2">
          <div>Left content</div>
          <div>Right content</div>
        </div>
      </DialogContent>,
    );

    expect(container).toMatchSnapshot();
  });
});
