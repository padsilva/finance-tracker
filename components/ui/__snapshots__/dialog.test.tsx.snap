// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Components should render basic dialog correctly 1`] = `
<div>
  <div
    data-testid="dialog-root"
  >
    <button
      data-testid="dialog-trigger"
    >
      Open Dialog
    </button>
    <div
      data-testid="dialog-portal"
    >
      <div
        class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
        data-testid="dialog-overlay"
      />
      <div
        class="fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg"
        data-testid="dialog-content"
      >
        <div
          class="flex flex-col space-y-1.5 text-center sm:text-left"
        >
          <h2
            class="text-lg font-semibold leading-none tracking-tight"
            data-testid="dialog-title"
          >
            Dialog Title
          </h2>
          <p
            class="text-sm text-muted-foreground"
            data-testid="dialog-description"
          >
            Dialog Description
          </p>
        </div>
        <div>
          Dialog Content
        </div>
        <div
          class="flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2"
        >
          <button
            data-testid="dialog-close"
          >
            Close Dialog
          </button>
        </div>
        <button
          class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
          data-testid="dialog-close"
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog Components should render dialog overlay with animations 1`] = `
<div>
  <div
    class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 custom-overlay"
    data-testid="dialog-overlay"
  />
</div>
`;

exports[`Dialog Components should render dialog with complex content 1`] = `
<div>
  <div
    data-testid="dialog-root"
  >
    <button
      data-testid="dialog-trigger"
    >
      Open Settings
    </button>
    <div
      data-testid="dialog-portal"
    >
      <div
        class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
        data-testid="dialog-overlay"
      />
      <div
        class="fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg"
        data-testid="dialog-content"
      >
        <div
          class="flex flex-col space-y-1.5 text-center sm:text-left"
        >
          <h2
            class="text-lg font-semibold leading-none tracking-tight"
            data-testid="dialog-title"
          >
            Settings
          </h2>
          <p
            class="text-sm text-muted-foreground"
            data-testid="dialog-description"
          >
            Manage your account settings and preferences.
          </p>
        </div>
        <div
          class="grid gap-4 py-4"
        >
          <div
            class="grid grid-cols-4 items-center gap-4"
          >
            <label
              for="name"
            >
              Name
            </label>
            <input
              class="col-span-3"
              id="name"
            />
          </div>
          <div
            class="grid grid-cols-4 items-center gap-4"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              class="col-span-3"
              id="username"
            />
          </div>
        </div>
        <div
          class="flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2"
        >
          <button
            data-testid="dialog-close"
          >
            Cancel
          </button>
          <button
            type="submit"
          >
            Save changes
          </button>
        </div>
        <button
          class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
          data-testid="dialog-close"
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog Components should render dialog with custom styling 1`] = `
<div>
  <div
    data-testid="dialog-root"
  >
    <button
      data-testid="dialog-trigger"
    >
      Open
    </button>
    <div
      data-testid="dialog-portal"
    >
      <div
        class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
        data-testid="dialog-overlay"
      />
      <div
        class="fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg custom-content"
        data-testid="dialog-content"
      >
        <div
          class="flex flex-col space-y-1.5 text-center sm:text-left custom-header"
        >
          <h2
            class="text-lg font-semibold leading-none tracking-tight custom-title"
            data-testid="dialog-title"
          >
            Custom Title
          </h2>
          <p
            class="text-sm text-muted-foreground custom-description"
            data-testid="dialog-description"
          >
            Custom Description
          </p>
        </div>
        <div
          class="flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2 custom-footer"
        >
          <button
            class="custom-close"
            data-testid="dialog-close"
          >
            Close
          </button>
        </div>
        <button
          class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
          data-testid="dialog-close"
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog Components should render dialog with different states 1`] = `
<div>
  <div
    data-testid="dialog-root"
  >
    <div
      data-testid="dialog-portal"
    >
      <div
        class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
        data-testid="dialog-overlay"
      />
      <div
        class="fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg"
        data-testid="dialog-content"
      >
        <h2
          class="text-lg font-semibold leading-none tracking-tight"
          data-testid="dialog-title"
        >
          Open State
        </h2>
        <button
          class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
          data-testid="dialog-close"
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
    </div>
    <div
      data-testid="dialog-portal"
    >
      <div
        class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
        data-testid="dialog-overlay"
      />
      <div
        class="fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg"
        data-testid="dialog-content"
      >
        <h2
          class="text-lg font-semibold leading-none tracking-tight"
          data-testid="dialog-title"
        >
          Closed State
        </h2>
        <button
          class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
          data-testid="dialog-close"
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog Components should render footer with different layouts 1`] = `
<div>
  <div
    class="flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2"
  >
    <button>
      Cancel
    </button>
    <button>
      Confirm
    </button>
  </div>
  <div
    class="flex sm:flex-row sm:justify-end sm:space-x-2 flex-row justify-between"
  >
    <button>
      Back
    </button>
    <div
      class="flex gap-2"
    >
      <button>
        Cancel
      </button>
      <button>
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog Components should render header with different alignments 1`] = `
<div>
  <div
    class="flex flex-col space-y-1.5 text-center sm:text-left"
  >
    <h2
      class="text-lg font-semibold leading-none tracking-tight"
      data-testid="dialog-title"
    >
      Default Centered Header
    </h2>
  </div>
  <div
    class="flex flex-col space-y-1.5 text-center sm:text-left"
  >
    <h2
      class="text-lg font-semibold leading-none tracking-tight"
      data-testid="dialog-title"
    >
      Left Aligned Header
    </h2>
  </div>
</div>
`;

exports[`Dialog Components should render responsive dialog content 1`] = `
<div>
  <div
    data-testid="dialog-portal"
  >
    <div
      class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
      data-testid="dialog-overlay"
    />
    <div
      class="fixed left-[50%] top-[50%] z-50 grid w-full translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg max-w-[90%] md:max-w-[80%] lg:max-w-[60%]"
      data-testid="dialog-content"
    >
      <div
        class="flex flex-col space-y-1.5 text-center sm:text-left"
      >
        <h2
          class="text-lg font-semibold leading-none tracking-tight"
          data-testid="dialog-title"
        >
          Responsive Dialog
        </h2>
      </div>
      <div
        class="grid grid-cols-1 gap-4 md:grid-cols-2"
      >
        <div>
          Left content
        </div>
        <div>
          Right content
        </div>
      </div>
      <button
        class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
        data-testid="dialog-close"
      >
        <svg
          class="h-4 w-4"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sr-only"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog Components should render with close button and icon 1`] = `
<div>
  <div
    data-testid="dialog-portal"
  >
    <div
      class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
      data-testid="dialog-overlay"
    />
    <div
      class="fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg"
      data-testid="dialog-content"
    >
      <h2
        class="text-lg font-semibold leading-none tracking-tight"
        data-testid="dialog-title"
      >
        Dialog with Close
      </h2>
      <p
        class="text-sm text-muted-foreground"
        data-testid="dialog-description"
      >
        Content here
      </p>
      <button
        data-testid="dialog-close"
      >
        <svg
          class="h-4 w-4"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sr-only"
        >
          Close
        </span>
      </button>
      <button
        class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
        data-testid="dialog-close"
      >
        <svg
          class="h-4 w-4"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sr-only"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;
